### PR TITLE
#172814148 User reset password

### PR DESCRIPTION
**Pull request by blaise82**
#### What does this PR do?
- allow a user to change his/her password

#### Description of Task to be completed?
- add send an email form
- add reset password form
- validate data 
- connect to redux and make backend calls 
- test the feature

#### How should this be manually tested?
- clone this repo
- navigate to this branch using git checkout `ft-password-reset-15-May-170947380`
- run `npm install`
- run `npm run dev` "to start the development server

#### What are the relevant pivotal tracker stories?
#170947380

#### Screenshots (if appropriate)
image

image

[finishes 170947380]

### CODE REVIEW

File: src/components/ResetPasswordForm.js [hunk 1]
```diff
@@ -0,0 +1,188 @@
...
+	}
+	validate = () => {
+		let isError = false;
+		const errors = {
+			passwordErrorStatus: false,
+			confirmPasswordErrorStatus: false
+		};
+		const regex = RegExp(/^(?=.*\d)(?=.*[a-z])(?=.*[A-Z])[a-zA-Z\d@$.!%*#?&]/);
+		// const results = this.state.password.match(/^(?=.*\d)(?=.*[a-z])(?=.*[A-Z])[a-zA-Z\d@$.!%*#?&]/);
+		// console.log(results);
```
Comment (AbdoulNuru) [inline src/components/ResetPasswordForm.js]: You can remove the commented lines if they are no longer used

**Review by AbdoulNuru — CHANGES_REQUESTED** [review]

File: src/languages/messages/fr.json [hunk 2]
```diff
@@ -61,5 +61,9 @@
   "acco_supplier": "Fournisseur d'hébergement",
   "travel_admin": "Administrateur de voyage",
   "travel_tmember": "Membre de l'équipe de voyage",
-  "select-role": "Vous devez sélectionner un rôle"
-}
+  "select-role": "Vous devez sélectionner un rôle",
+  "confirm_password": "Confirmez le mot de passe",
```
Comment (AbdoulNuru) [inline src/languages/messages/fr.json:65]: Add new lines

File: src/components/ResetPasswordForm.js (hunk 1 above)
Comment (blaise82) [reply src/components/ResetPasswordForm.js]: done

File: src/languages/messages/fr.json (hunk 2 above)
Comment (blaise82) [reply src/languages/messages/fr.json:65]: done

**Review by Moise1 — CHANGES_REQUESTED** [review]
Hey @blaise82 , please write a proper PR description.

**Comment by blaise82** [comment]
> Hey @blaise82 , please write a proper PR description.

done

**Review by hakoemmy — COMMENTED** [review]
@blaise82 So far so good, however this what 
Screen Shot 2020-05-18 at 10 47 23
 I get when I click on ``` Click the link below to verify your password```. I guess it's a matter of changing a redirect URL env in the backend to point to the front end review app, give a short turnaround, and keep us in a loop when it's done.

In case you have any concern, please let me know.


I did mention this in the group

**Comment by Moise1** [comment]
@blaise82 , I think having two **Reset Your Password** identical titles and subtitles is a wrong duplication. Please revise that.

**Review by AbdoulNuru — APPROVED** [review]

**Review by Moise1 — APPROVED** [review]
Awesome job.

**Review by hakoemmy — APPROVED** [review]

File: src/styles/entry.module.css [hunk 3]
```diff
@@ -25,6 +25,13 @@ h1 {
     margin-right: auto;
     margin-left: auto;
 }
+.wrap2 {
```
Comment (AJAkimana) [inline src/styles/entry.module.css]: Give a descriptive name

File: src/styles/signup.module.css [hunk 4]
```diff
@@ -69,6 +69,11 @@
   margin-bottom: 30px;
   float: left;
 }
+.link2 {
```
Comment (AJAkimana) [inline src/styles/signup.module.css]: Give a descriptive name

File: src/languages/messages/en.json [hunk 5]
```diff
@@ -61,5 +58,10 @@
...
   "travel_tmember": "Travel team member",
-  "select-role": "You must select a role"
-}
+  "select-role": "You must select a role",
+  "confirm_password": "Confirm Password",
+  "password_reset": "Your password was reset successfully",
+  "sign-in-click": "Please click to sign in",
+  "password_reset_email": "We’ve emailed you instructions for setting a new password if an account exists with the email you entered. You should receive them shortly.",
+  "reset-email": "Get Reset Email"
+}
```
Comment (AJAkimana) [inline src/languages/messages/en.json]: Add new line

**Review by AJAkimana — COMMENTED** [review]
Work on the inline comments

**Review by AJAkimana — APPROVED** [review]

**Pull request merged**